### PR TITLE
Domino fit example

### LINKED ISSUE #163: Domino Benchmark
The goal of this issue is to implement this benchmark:

https://pedtsr.ca/2024/solving-domino-fit-using-constraint-programming.html

It might be implemented in two approaches: either by considering the constraints as fitness functions, or by encoding it as dependent types (or a mix of both).

### PR DESCRIPTION
**Pull request by Su2001**
Closes #163.

### CODE REVIEW

**Comment by alcides** [comment]
@Su2001 This is not passing the Ruff test. My guess is that you have not installed the pre-commit on your machine.

Furthermore, can you include the Domino example in the run_examples script?

**Comment by alcides** [comment]
Code review:

If you want to have different strategies to solve the problem, you should have each in a different file, as different benchmarks.


Regarding the fitness functions, they are too complex. Try to assess the following:

* Move part of the work to methods within the Domino classes.
* Try to use functional programming concepts (you want to fold/reduce over a list of pieces, passing a state of the board. That state dataclass can contain the matrix, as well as the conflicts.
* The fitness function creates the original board, passes it over to the solution, and extracts the relevant metrics of the final state.

File: examples/domino.py [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+    result='\n'
+    for i in visited:
+        for j in i:
+            result += str(j)+" "
+        result += "\n"
+    return result
+
+def toboardv3(board):
+    visited = generateMatrix(7)
+    for i in blacks:
```
Comment (GUIpsp) [inline examples/domino.py:265]: Probably should use de structuring for: ```for x, y in blacks:```

**Comment by GUIpsp** [comment]
Some things seem to not be properly formatted still, you should run pre-commit run --all, or ruff fmt

**Comment by Su2001** [comment]
I have run pre-commit run --all, but it found an the error in this file tests/representations/basic_representation_test.py and no more fails are found

**Pull request merged**